base_layer/core/tests/mod.rs

```diff
@@ -0,0 +1,27 @@
+//  Copyright 2019 The Tari Project
```

Comment:
I don't know if this is what we want.

According to [the rust book](https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests),

* Every top level file in `tests` is its own "crate"; i.e. an independent set of integration tests
* `mods` inside `test` are not considered tests, but are treated as helper modules that the test "crates" can use.

> Files in subdirectories of the tests directory don’t get compiled as separate crates or have sections in the test output.

Tests _seem_ to run with this re-arrangement, but it definitely seems to go against what the book says

Reply:
I reverted the changes and moved `horizon_state_validators.rs` a level up. The approach I used was similar to what was used in tari_comms to group related tests. When we have tests from many different subsystems, I think it can look cleaner.

Reply:
👍